### DISCUSSION TITLE
Autostarting xwayland apps without setting workspace to urgent

### DISCUSSION THREAD

**Discussion by raphaelweis** [hyprwm/Hyprland#6716] [Q&A]
I autostart some apps when launching hyprland using exec-once. This works fine for wayland apps such as firefox and alacritty, but I noticed that xwayland apps set the urgency of the workspace when starting, causing the workspace icon from the waybar hyprland module to turn red. Is there a way to disable this behavior only on startup ? I tried searching the issues, discussions and wiki but didn't find anything, maybe I'm missing something...

For reference, this is how I autostart my apps, using the home-manager hyprland module :
```nix
exec-once = [
  # firefox and alacritty work fine, they autostart silently without turning the workspace icon red
  "[workspace 1 silent] ${pkgs.firefox}/bin/firefox"
  "[workspace 2 silent] ${pkgs.alacritty}/bin/alacritty"
  
  # discord and spotify both set the workspace icon to red on startup
  "[workspace 4 silent] ${pkgs.discord}/bin/discord"
  "[workspace 5 silent] ${pkgs.spotify}/bin/spotify"
];
```

**Comment by winkelnp** [comment]
This happens to my config with (wayland) firefox and thunderbird, but it seems only when firefox wants to (and fails to) restore the session or thunderbird change tab (as in mail in place of calendar). I'm not sure how to fix anything, but I do think there should be a way to supress urgent events. 

One workaround is to add a delay to the autostarting of waybar, for me just adding `sleep 3; waybar` was enough.
